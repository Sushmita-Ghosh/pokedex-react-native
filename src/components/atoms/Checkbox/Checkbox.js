import { View, Text, StyleSheet } from "react-native";
import React, { useState, memo, useEffect } from "react";
import { CheckBox } from "@rneui/themed";

const Checkbox = ({ item, toggleCheckbox }) => {
  return (
    <View
      style={{
        width: "50%",
      }}
    >
      <CheckBox
        checked={item.isChecked}
        onPress={() => toggleCheckbox(item.name)}
        // Use ThemeProvider to make change for all checkbox
        iconType="material-community"
        checkedIcon="checkbox-marked"
        uncheckedIcon="checkbox-blank-outline"
        checkedColor="black"
        title={item.name}
        textStyle={{
          fontSize: 18,
          textTransform: "capitalize",
        }}
      />
    </View>
  );
};

export default memo(Checkbox);
