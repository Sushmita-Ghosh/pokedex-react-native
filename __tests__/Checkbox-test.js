import React from "react";
import Checkbox from "../src/components/atoms/Checkbox/Checkbox";
import { render, screen, fireEvent } from "@testing-library/react-native";

test("Check if CheckBox rendered Correctly", () => {
  const mockBaseProps = {
    item: {},
    toggleCheckbox: jest.fn(),
  };

  const Component = (props) => <Checkbox {...mockBaseProps} {...props} />;
  render(
    <Component
      item={{
        name: "fire",
        url: "https://pokeapi.co/api/v2/type/10/",
        isChecked: false,
      }}
    />
  );

  expect(screen.toJSON()).toMatchSnapshot();
});
