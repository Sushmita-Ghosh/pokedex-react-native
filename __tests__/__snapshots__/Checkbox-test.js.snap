// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if CheckBox rendered Correctly 1`] = `
<View
  style={
    {
      "width": "50%",
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      {
        "busy": undefined,
        "checked": false,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    checkedIcon="checkbox-marked"
    collapsable={false}
    focusable={true}
    iconType="material-community"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#ffffff",
        "margin": 5,
        "marginLeft": 10,
        "marginRight": 10,
        "padding": 10,
      }
    }
    testID="RNE__CheckBox__Wrapper"
    uncheckedIcon="checkbox-blank-outline"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#bfbfbf",
              "fontSize": 24,
            },
            {
              "minWidth": 24,
            },
            {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="RNE__Checkbox__Icon"
      >
        ó°„±
      </Text>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#43484d",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginLeft": 10,
            "marginRight": 10,
            "textTransform": "capitalize",
          }
        }
        testID="RNE__CheckBox__Title"
      >
        fire
      </Text>
    </View>
  </View>
</View>
`;
